tenseal/cpp/tensors/ckkstensor.cpp

```diff
@@ -72,17 +81,33 @@ PlainTensor<double> CKKSTensor::decrypt(const shared_ptr<SecretKey>& sk) const {
     Decryptor decryptor =
         Decryptor(*this->tenseal_context()->seal_context(), *sk);
 
-    vector<double> result;
-    vector<double> buff;
-    result.reserve(this->_data.size());
-
-    for (size_t i = 0; i < this->_data.size(); i++) {
-        decryptor.decrypt(this->_data[i], plaintext);
-        this->tenseal_context()->decode<CKKSEncoder>(plaintext, buff);
-        result.push_back(buff[0]);
+    if (_batching_enabled) {
+        vector<vector<double>> result;
+        result.reserve(this->_data.size());
+
+        for (size_t i = 0; i < this->_data.size(); i++) {
+            vector<double> buff;
+            decryptor.decrypt(this->_data[i], plaintext);
+            this->tenseal_context()->decode<CKKSEncoder>(plaintext, buff);
+            result.push_back(
+                vector<double>(buff.begin(), buff.begin() + _shape[0]));
+        }
+        return PlainTensor<double>(/*batched_tensor=*/result,
+                                   /*original_shape=*/_shape, /*batch_axis=*/0);
```

Comment:
The strides are directly generated from the shape, no? Every PlainTensor constructor calls generate_strides(shape).

Reply:
Yeah, but as tensor might be reshaped, which might incur changes to the strides, using the generated strides may be false. The generate strides will generate default strides, but as tensor are reshaped, the value might change (instead of moving data in the vector)

Reply:
Yes, I was saying, we can recompute the strides whenever needed, to prevent bugs. But the stride is always derived from the shape, even from the reshaping.

Reply:
But I don't understand the issue here: how can the strides be changed right now? other than via the constructor

Reply:
I removed the strides member, until we will actually need it.